x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/forcemerge.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiDescribedFormGroup,
+  EuiFieldNumber,
+  EuiSpacer,
+  EuiSwitch,
+  EuiTextColor,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { LearnMoreLink } from './learn_more_link';
+import { ErrableFormRow } from './form_errors';
+import { Phases, PhaseWithForcemergeAction } from '../../../../../common/types';
+import { PhaseValidationErrors, propertyof } from '../../../services/policies/policy_validation';
+
+const forcemergeLabel = i18n.translate('xpack.indexLifecycleMgmt.warmPhase.forceMergeDataLabel', {
+  defaultMessage: 'Force merge data',
+});
+
+interface Props<T extends PhaseWithForcemergeAction> {
+  errors?: PhaseValidationErrors<T>;
+  phase: keyof Phases & string;
+  phaseData: T;
+  setPhaseData: (dataKey: keyof T & string, value: boolean | string) => void;
+  isShowingErrors: boolean;
+}
+export const Forcemerge = <T extends PhaseWithForcemergeAction>({
```

Comment:
If we remove the generic we can also type this as:

```ts
export const Forcemerge: FunctionComponent<Props> = ({ ... }) => { ...
```

and remove `React.PropsWithChildren<..>`  below.